cmd/lakectl/cmd/local_pull.go

```diff
@@ -63,8 +70,11 @@ var localPullCmd = &cobra.Command{
 			}
 			return nil
 		})
-		s := local.NewSyncManager(cmd.Context(), client, syncFlags.parallelism, syncFlags.presign)
+		s := local.NewSyncManager(ctx, client, syncFlags.parallelism, syncFlags.presign)
```

Comment:
Consider passing the sigCtx only to the SyncManager and not for other operations

Reply:
I considered that, but in such case - an interruption will not be handled during the preliminary requests.

Reply:
If you put the New right before the Sync call - you could potentially use this context less

Reply:
Also if you use the suggested method with the function - we will call Stop() immediately after receiving the first signal and revert the signal override